Assets/Scripts/Scheduler/Scheduler.cs

```diff
@@ -0,0 +1,241 @@
+
+        public Dictionary<string, EventPrototype> EventPrototypes { get; protected set; }
+
+        #region LuaHandling
+
+        public static void LoadScripts()
+        {
+            string luaFilePath = Path.Combine(Application.streamingAssetsPath, "LUA");
+            luaFilePath = Path.Combine(luaFilePath, "Events.lua");
+            LuaUtilities.LoadScriptFromFile(luaFilePath);
+        }
+
+        public static void LoadModsScripts(DirectoryInfo[] mods)
+        {
+            foreach (DirectoryInfo mod in mods)
+            {
+                string luaModFile = Path.Combine(mod.FullName, "Events.lua");
+                if (File.Exists(luaModFile))
+                {
+                    LuaUtilities.LoadScriptFromFile(luaModFile);
+                }
+            }
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Schedules an event from a prototype.
+        /// </summary>
+        /// <param name="name">Name of the event (for serialization etc.).</param>
+        /// <param name="cooldown">Cooldown in seconds.</param>
+        /// <param name="repeatsForever">Whether the event repeats forever (defaults false). If true repeats is ignored.</param>
+        /// <param name="repeats">Number of repeats (default 1). Ignored if repeatsForever=true.</param>
+        public void ScheduleEvent(string name, float cooldown, bool repeatsForever = false, int repeats = 1)
+        {
+            if (EventPrototypes.ContainsKey(name) == false)
+            {
+                Debug.ULogWarningChannel("Scheduler", "Tried to schedule an event from a prototype '{0}' which does not exist. Bailing.", name);
+                return;
+            }
+
+            EventPrototype ep = EventPrototypes[name];
+            ScheduledEvent evt = new ScheduledEvent(ep, cooldown, cooldown, repeatsForever, repeats);
+
+            RegisterEvent(evt);
+        }
+
+        public void RegisterEvent(ScheduledEvent evt)
+        {
+            if (evt != null)
+            {
+                if (IsRegistered(evt))
+                {
+                    Debug.ULogChannel("Scheduler", "Event '{0}' registered more than once.", evt.Name);
+                }
+
+                Events.Add(evt);
+            }
+        }
+
+        public bool IsRegistered(ScheduledEvent evt)
+        {
+            return Events != null && Events.Contains(evt);
+        }
+
+        /// <summary>
+        /// Deregisters the event.
+        /// NOTE: To stop a continuing event from running do not try to deregister it with this!
+        /// Instead call Stop() on the event. Then it will be removed on the next purge.
+        /// </summary>
+        /// <param name="evt">Event to deregister.</param>
+        public void DeregisterEvent(ScheduledEvent evt)
+        {
+            if (Events != null)
+            {
+                Events.Remove(evt);
+            }
+        }
+
+        public void PurgeEventList()
```

Comment:
Rename to `RemoveFinishedEvents` or `ClearFinishedEvents`
By name `PurgeEventList` I decided that you are going to clear the whole list